src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -43,42 +57,112 @@ public Intake(int motorId, int frontPiston, int backPiston) {
+        tiltEncoder = tiltMotor.getEncoder();
+        tiltController = tiltMotor.getClosedLoopController();
+        tiltEncoder.setPosition(0);
+
+
+        SmartDashboard.putBoolean("Passive intake out", false);
+    }
+
+    public void prioState() {
+        if (SmartDashboard.getBoolean("Passive intake out", false) & passiveIntakePrio == PassiveIntake.IN) {
+            passiveIntakePrio = PassiveIntake.OUT;
+        }
+        else if (SmartDashboard.getBoolean("Passive intake out", true) & passiveIntakePrio == PassiveIntake.OUT){
+            passiveIntakePrio = PassiveIntake.IN;
+        }
+    }
+
+    public SequentialCommandGroup grabCoralSequence() {
+        return new SequentialCommandGroup(
+            intakeOut(), grabCoral()
+        );
+    }
 
-        intakeModule = new PneumaticsControlModule(0);
+    public SequentialCommandGroup releaseCoralSequence() {
+        return new SequentialCommandGroup(
+           stopGrab(), intakeIn(), releaseCoral(), new WaitCommand(.5), stopRelease()
+        );
+    }
+    public  Command intakeMid() {
+        return Commands.runOnce(()-> {
+            tiltController.setReference(Constants.IntakeConstants.EXTENSION_MID, SparkMax.ControlType.kPosition);
+            intakePosition = IntakePosition.MID;
+        });
+    }
 
-        intakeSolenoid = new Solenoid(0, PneumaticsModuleType.CTREPCM, frontPiston);
-        intakeSolenoid.set(false); //might be needed to be set to true
+    public Command intakeOut() {
+        return Commands.runOnce(()-> {
+            tiltController.setReference(Constants.IntakeConstants.EXTENSION_OUT, SparkMax.ControlType.kPosition);
+            intakePosition = IntakePosition.OUT;
+        });
     }
 
-    public Command toggleSolenoid() {
-        return Commands.runOnce(() -> {
-            if (solenoidState == SolenoidState.FALSE) {
-                intakeSolenoid.set(true);
-                solenoidState = SolenoidState.TRUE;
-            } else {
-                intakeSolenoid.set(false);
-                solenoidState = SolenoidState.FALSE;
+    public Command intakeIn() {
+        return Commands.runOnce(()-> {
+            tiltController.setReference(Constants.IntakeConstants.EXTENSION_IN, SparkMax.ControlType.kPosition);
+            intakePosition = IntakePosition.IN;
+        });
+    }
+    
+    public Command stopGrab() {
+        return Commands.runOnce(()-> {
+            grabMotor.set(0);
+            CommandScheduler.getInstance().cancel(grabCoral());
+        });
+    }
+
+    public Command stopRelease() {
+        return Commands.runOnce(()-> {
+            grabMotor.set(0);
+            CommandScheduler.getInstance().cancel(releaseCoral());
+        });
+    }
+
+    public Command grabCoral() {
+        return Commands.run(() -> {
+            grabMotor.set(.5);
+            intakeState = IntakeState.FULL;
+            while (true) {
```

Comment:
nepoužívat While a For struktury! Ty zacyklí robota v nekonečném cyklu a spadne to. Tady stačí jednoduchý if() statement. Vzhledem k iterativní charakteristice programu se chová úplně stejně jako while()